### PR TITLE
Add support for `set_parent` in XDG portals

### PR DESCRIPTION
**Pull request by kjarosh**
Related to https://github.com/PolyMeilex/rfd/issues/46.

This PR ads support for `set_parent` for XDG Portals. As mentioned in https://github.com/PolyMeilex/rfd/issues/46, that requires making the parent to be `HasWindowHandle + HasDisplayHandle` due to `WindowIdentifier` on Wayland requiring a `RawDisplayHandle`.

Currently the code calls `block_on` when constructing a `WindowIdentifier` due to the fact that the handles are not `Send`, but it's much better compared to not having the feature at all \:)

Tested on X11.

### CODE REVIEW

File: src/backend/xdg_desktop_portal.rs [hunk 1]
```diff
@@ -7,11 +7,22 @@ use crate::message_dialog::MessageDialog;
...
+fn to_window_identifier(
+    window: Option<RawWindowHandle>,
+    display: Option<RawDisplayHandle>,
+) -> Option<WindowIdentifier> {
+    window.map(|window| {
+        block_on(Box::pin(async move {
+            WindowIdentifier::from_raw_handle(&window, display.as_ref()).await
+        }))
+    })
+}
```
Comment (PolyMeilex) [inline src/backend/xdg_desktop_portal.rs:25]: Could we drop the block_on and make `to_window_identifier` async instead?
Every call-site of `to_window_identifier` is in async context already anyway.
```suggestion
async fn to_window_identifier(
    window: Option<RawWindowHandle>,
    display: Option<RawDisplayHandle>,
) -> Option<WindowIdentifier> {
    let identifier = WindowIdentifier::from_raw_handle(&window?, display.as_ref()).await;
    Some(identifier)
}
```

**Review by PolyMeilex — APPROVED** [review]

**Comment by kjarosh** [comment]
Thank you for your review! I've added installing `libwayland-dev` as a dependency for Ubuntu XDG in CI hoping it will fix the failure due to missing wayland-client system library

**Comment by PolyMeilex** [comment]
Thanks!

**Comment by torokati44** [comment]
Is there maybe a timeline for when a release with this feature included in it is planned to be released? 👀😊

**Pull request merged**